drivers/hmis/app/models/hmis/active_range.rb

```diff
@@ -6,9 +6,28 @@
 
 class Hmis::ActiveRange < Hmis::HmisBase
   self.table_name = :hmis_active_ranges
+  include ::Hmis::Concerns::HmisArelHelper
   belongs_to :entity, polymorphic: true, optional: true
+  belongs_to :user, class_name: 'Hmis::User'
 
-  def self.for_entity(entity)
-    Hmis::ActiveRange.where(entity: entity).order(:updated_at).last
+  # Each UnitOccupancy can only have 1 ActiveRange
+  validates_uniqueness_of :entity_id, scope: :entity_type, conditions: -> { where(entity_type: Hmis::UnitOccupancy.name) }
```

Comment:
@eanders a nice solution, I thought! `scope: :entity_type` isn't _really_ necessary here, but I'm leaving it so that we can add in additional types to the `where` clause and still have it work.